src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.Workable;
+import core.mate.academy.service.producers.BulldozerProducer;
+import core.mate.academy.service.producers.ExcavatorProducer;
+import core.mate.academy.service.producers.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer<? extends Machine> someProducer = null;
+        if (type.equals(Bulldozer.class)) {
+            someProducer = new BulldozerProducer();
+        } else if (type.equals(Excavator.class)) {
+            someProducer = new ExcavatorProducer();
+        } else if (type.equals(Truck.class)) {
+            someProducer = new TruckProducer();
+        }
```

Comment:
I think all producers should be initialized like instances in class and after used in method